### ISSUE TITLE
Specifying `target` all the time in polygon methods is painful

### ISSUE THREAD

**Issue by asinghvi17** [asinghvi17/GeometryOps.jl#66]
Can we have it default to the greater of the two types given?

**Comment by skygering** [comment]
I don't think so. Because if you give two `Polygons`, in LibGEOS it can return anything from a `Point` to a `LineString` to a `Polygon`. I think default behavior should probably mimic LibGEOS, which was my plan in the long term (but a list instead of a `GeometryCollection`). However, we could change that to do what you have suggested. It is more applicable for my use case. But then if people want LibGEOS behavior they have to specify that they want a `Union{Point, Line, LineString, Polygon} `which seems more annoying to me.

**Comment by asinghvi17** [comment]
Ah, I see what you're saying.  I guess I wanted unspecified `target` to give the algorithm freedom to return anything, so the user doesn't have to specify at all.

**Comment by skygering** [comment]
Ah yes. I agree with that. That is the long-term plan. So the default target would be` Union{GI.Point, GI.Line, GI.LineString, GI.Polygon} ` it is just that doesn't work right now, so I didn't want to give people a false understanding of what is possible with the current code.
